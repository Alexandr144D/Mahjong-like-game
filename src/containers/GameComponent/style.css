.game-board {
    flex-direction: row;
    flex-wrap: wrap;
    display: flex;
    width: 504px;
    animation-name: boardAppear;
    animation-timing-function: ease;
    animation-duration: .2s;
    padding: 4px;
    border-radius: 4px;
    background: rgba(0, 0, 0, 0.1);
}

@keyframes boardAppear {
    0% {
        transform: translateY(-60px) scaleY(0);
        transform-origin: top;
        opacity: 0
    }
    100% {
        transform: translateY(0) scaleY(1);
        transform-origin: top;
        opacity: 1
    }
}

.react-icon {
    width: 30px;
    height: 30px;
}

h3 {
    position: absolute;
}

.new-game-btn {
    width: 170px;
    position: absolute;
    bottom: -20px;
    left: calc(50% - 85px);
}
