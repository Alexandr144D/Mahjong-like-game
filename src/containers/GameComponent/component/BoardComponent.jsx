import '../style.css';
import React from 'react';
import PropTypes from 'prop-types';
import Paper from '../../../components/PaperComponent';


const BoardComponent = (props) => {
    return (
        <div className='game-board'>
            {
                props.rootReducer.gameBoard.map((item, index) => {
                    return (
                        <Paper
                            {...props}
                            key={index}
                            item={item}
                            index={index}
                            checked={item.checked}
                        />)
                })
            }
        </div>
    )
};

BoardComponent.propTypes = {
    props: PropTypes.object,
};

export default BoardComponent;
