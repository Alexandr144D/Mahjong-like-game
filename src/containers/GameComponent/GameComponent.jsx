import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import Button from '@material-ui/core/Button';
import BoardComponent from './component/BoardComponent';
import {updateTable, newGameStartAction} from '../../reducer/actions'


class GameComponent extends React.Component {
    constructor(props) {
        super(props);

        this.newGame = this.newGame.bind(this);
    }

    componentDidMount() {
        if (!this.props.rootReducer.gameBoard.length) {
            this.props.history.push('/entry')
        }
    }

    upDateBoardState = (index) => {
        this.props.onUpdateBoardDispatch(index)
    };

    newGame() {
        this.props.onStartNewGameDispatch()
    }

    render() {
        const {isNewGameButtonVisible} = this.props.rootReducer;
        return (
            <div>
                <BoardComponent upDateBoardState={this.upDateBoardState} {...this.props} />
                {
                    isNewGameButtonVisible &&
                    <Button
                        color="primary"
                        variant="contained"
                        className='new-game-btn'
                        onClick={this.newGame}
                    >
                        New Game
                    </Button>
                }
            </div>
        )
    }
}

const mapStateToProps = state => {
    return state
};

const mapDispatchToProps = (dispatch) => {
    return {
        onUpdateBoardDispatch: (index) => {
            dispatch(updateTable(index))
        },
        onStartNewGameDispatch: () => {
            dispatch(newGameStartAction())
        }
    }
};

GameComponent.propTypes = {
    newGame: PropTypes.func,
    state: PropTypes.object,
    updateTable: PropTypes.func,
    upDateBoardState: PropTypes.func,
    newGameStartAction: PropTypes.func,
    isNewGameButtonVisible: PropTypes.bool,
    onUpdateBoardDispatch: PropTypes.func.isRequired,
    onStartNewGameDispatch: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(GameComponent);
