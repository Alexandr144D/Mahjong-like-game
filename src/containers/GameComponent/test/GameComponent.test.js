import React from 'react';
import {shallow, mount, render} from 'enzyme';
import GameComponent from '../GameComponent';
import BoardComponent from '../../GameComponent/component/BoardComponent';
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";


const mockStore = configureMockStore();

describe("GameComponent", () => {

    let store;
    beforeEach(() => {
        store = mockStore({
            rootReducer: {
                gameBoard: [],
                chosenCell1: undefined,
                chosenCell2: undefined,
                isNewGameButtonVisible: false
            },
            history: {
                push: jest.fn()
            }
        });
    });

    it("should render without throwing an error", () => {
        const store = mockStore({
            rootReducer: {
                gameBoard: [],
                chosenCell1: undefined,
                chosenCell2: undefined,
                isNewGameButtonVisible: true
            },
            history: {
                push: jest.fn()
            }
        });
        const wrapper = mount(
            <Provider store={store}>
                <GameComponent/>
            </Provider>
        );

        expect(wrapper).toMatchSnapshot()
    });

    it("should render without throwing an error", () => {
        expect(shallow(<Provider store={store}><GameComponent/></Provider>)).toBeDefined()
    });

    it("should render without throwing an error", () => {
        const store = mockStore({
            rootReducer: {
                gameBoard: [],
                chosenCell1: undefined,
                chosenCell2: undefined,
                isNewGameButtonVisible: true
            },
            history: {
                push: jest.fn()
            }
        });
        const wrapper = mount(
            <Provider store={store}>
                <GameComponent/>
            </Provider>
        );
        expect(wrapper.find('BoardComponent').length).toEqual(1)
    });

    it("should show 'New game Button' when the game is finished", () => {
        const store = mockStore({
            rootReducer: {
                gameBoard: [],
                chosenCell1: undefined,
                chosenCell2: undefined,
                isNewGameButtonVisible: true
            },
            history: {
                push: jest.fn()
            }
        });

        const wrapper = mount(
            <Provider store={store}>
                <GameComponent/>
            </Provider>
        );

        expect(wrapper.find('button').length).toEqual(1);
        expect(wrapper.find('button').text()).toEqual('New Game')
    });

    it("should show 'New game Button' when the game is finished", () => {
        const store = mockStore({
            rootReducer: {
                gameBoard: [],
                chosenCell1: undefined,
                chosenCell2: undefined,
                isNewGameButtonVisible: false
            },
            history: {
                push: jest.fn()
            }
        });

        const wrapper = mount(
            <Provider store={store}>
                <GameComponent/>
            </Provider>
        );

        expect(wrapper.find('button').length).toEqual(0);
    });

});
