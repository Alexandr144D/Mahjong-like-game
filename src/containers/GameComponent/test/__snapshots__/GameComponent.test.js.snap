// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameComponent should render without throwing an error 1`] = `ReactWrapper {}`;
